### PR TITLE
Fixed #26651 -- Kept original file suffix in TemporaryUploadedFile name

### PR DESCRIPTION
**Pull request by claudep**
https://code.djangoproject.com/ticket/26651

### CODE REVIEW

File: django/core/files/uploadedfile.py [hunk 1]
```diff
@@ -58,7 +58,8 @@ class TemporaryUploadedFile(UploadedFile):
     A file uploaded to a temporary location (i.e. stream-to-disk).
     """
     def __init__(self, name, content_type, size, charset, content_type_extra=None):
-        file = tempfile.NamedTemporaryFile(suffix='.upload', dir=settings.FILE_UPLOAD_TEMP_DIR)
+        ext = os.path.splitext(name)[1]
```
Comment (timgraham) [inline django/core/files/uploadedfile.py]: I think: `_, ext = os.path.splitext(name)` is a bit more readable than "magical" indexing.

File: tests/files/tests.py [hunk 2]
```diff
@@ -214,6 +215,13 @@ def test_open_resets_file_to_start_and_returns_context_manager(self):
             self.assertEqual(f.read(), '1')
 
 
+class TemporaryUploadedFileTests(unittest.TestCase):
+    def test_extension_kept(self):
+        """The temporary file name has same suffix as original file."""
```
Comment (timgraham) [inline tests/files/tests.py]: the same suffix as the original file

**Comment by jdufresne** [comment]
Something to recognize, `os.path.splitext()` splits on the right most `'.'`. Sometimes this may not be what one would normally consider the file suffix. e.g. one might consider `.tar.gz` the real suffix for the file below:

```python
>>> os.path.splitext('django.tar.gz')
('django.tar', '.gz')
```

I don't really have a good solution to workaround this and it may not be an issue for the intended use case, but I thought it should be noted.

**Comment by timgraham** [comment]
https://code.djangoproject.com/ticket/23759 notes the same issue elsewhere in Django.

**Comment by claudep** [comment]
I don't think it's really crucial here, but if we find an elegant solution to 23579, we can then apply it here too.

**Pull request closed without merging**